### PR TITLE
Add support for empty to-one relationship according to JSON API v1.0

### PR DESCRIPTION
**Pull request by RafaelKayumov**
Example from spec here: http://jsonapi.org/format/#fetching-relationships-responses

Tests also added

### CODE REVIEW

File: Project/JSONAPITests/JSONAPITests.m [hunk 1]
```diff
@@ -239,8 +239,27 @@ - (void)testGenericSerialization {
...
 
+#pragma mark - Empty relationship tests
+
+- (void)testEmptyRelationship {
+    NSDictionary *json = [self emptyRelationshipsExampleJSON];
+    JSONAPI *jsonAPI = [JSONAPI jsonAPIWithDictionary:json];
+    
+    NewsFeedPostResource *testPost = jsonAPI.resource;
+    
+    XCTAssertNil(testPost.publisher, @"Test post's attachments should not be nil");
```
Comment (tarbrain) [inline Project/JSONAPITests/JSONAPITests.m]: Small copy-and-paste error: the description should read `Test post's publisher should be nil`.

**Comment by tarbrain** [comment]
Looks good and useful to me. I think we should include it.

**Comment by RafaelKayumov** [comment]
@joshdholtz ping

**Comment by joshdholtz** [comment]
@RafaelKayumov Thanks for the ping :blush: Merging and stuff

**Pull request merged**